auth0/src/test/java/com/auth0/android/result/CredentialsTest.kt

```diff
@@ -0,0 +1,38 @@
+package com.auth0.android.result
+
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.Test
+import java.util.*
+
+public class CredentialsTest {
+    @Test
+    public fun shouldCreate() {
+        val date = Date()
+        val credentials: Credentials =
+            CredentialsMock("idToken", "accessToken", "type", "refreshToken", date, "scope")
+        MatcherAssert.assertThat(credentials.idToken, Matchers.`is`("idToken"))
+        MatcherAssert.assertThat(credentials.accessToken, Matchers.`is`("accessToken"))
+        MatcherAssert.assertThat(credentials.type, Matchers.`is`("type"))
+        MatcherAssert.assertThat(credentials.refreshToken, Matchers.`is`("refreshToken"))
+        MatcherAssert.assertThat(credentials.expiresAt, Matchers.`is`(date))
+        MatcherAssert.assertThat(credentials.scope, Matchers.`is`("scope"))
+        MatcherAssert.assertThat(credentials.recoveryCode, Matchers.`is`(Matchers.nullValue()))
+    }
+
+    @Test
+    public fun shouldGetScope() {
+        val credentials =
+            Credentials("idToken", "accessToken", "type", "refreshToken", Date(), "openid profile")
+        MatcherAssert.assertThat(credentials.scope, Matchers.`is`("openid profile"))
+    }
+
+    @Test
+    public fun shouldGetRecoveryCode() {
+        val date = Date()
+        val credentials: Credentials =
+            CredentialsMock("idToken", "accessToken", "type", "refreshToken", date, "scope")
```

Comment:
Shouldn't `recoveryCode` be passed to the constructor?

Reply:
There is no new constructor that takes a recovery code. I didn't want to create a secondary constructor just for this one property, and changing the existing one would be a breaking change even when default values are used. Kotlin doesn't enforce the use of "named arguments" so anyone could be calling it the way we are calling it in this specific line and would be facing a breaking change due to a missing property.